rpc/client/evidence_test.go

```diff
@@ -171,3 +168,19 @@ func TestBroadcastEmptyEvidence(t *testing.T) {
 		assert.Error(t, err)
 	}
 }
+
+func waitForBlock(t *testing.T, ctx context.Context, c client.Client, height int64) {
+	for {
+		status, err := c.Status(ctx)
+		require.NoError(t, err)
+		if status.SyncInfo.LatestBlockHeight >= height {
+			return
+		}
+		timer := time.NewTimer(200 * time.Millisecond)
+		select {
+		case <-ctx.Done():
+			return
+		case <-timer.C:
```

Comment:
I'd put lines 174-83 here, creating the timer outside of the for loop (with a defer timer.Stop()) but initializing the timer at 10 and only resetting it to 200ms.

Reply:
Good catch. We should use a single Timer with a defer'd stop to not leak resources.

Reply:
I'm not sure if I'm full comprehending your suggestion. Do you mean something like this:
```
func waitForBlock(t *testing.T, ctx context.Context, c client.Client, height int64) {
	ticker := time.NewTicker(200 * time.Millisecond)
	defer ticker.Stop()
	for {
		select {
		case <-ctx.Done():
			return
		case <-ticker.C:
			status, err := c.Status(ctx)
			require.NoError(t, err)
			if status.SyncInfo.LatestBlockHeight >= height {
				return
			}
		}
	}
}
```

Reply:
yeah, exactly, though I'd use a timer that's initalized to 0, and then reset it at the end of the case statement, so that it fires immediately, and not make *every* caller wait 200ms even if the block exists when the function is called.

Reply:
```
func waitForBlock(ctx context.Context, t *testing.T, c client.Client, height int64) {
	timer := time.NewTimer(0 * time.Millisecond)
	defer timer.Stop()
	for {
		select {
		case <-ctx.Done():
			return
		case <-timer.C:
			status, err := c.Status(ctx)
			require.NoError(t, err)
			if status.SyncInfo.LatestBlockHeight >= height {
				return
			}
			timer.Reset(200 * time.Millisecond)
		}
	}
}
```
Right. More like this?